shared/src/main/scala/kinesis4cats/producer/fs2/FS2Producer.scala

```diff
@@ -86,6 +86,47 @@ abstract class FS2Producer[F[_], PutReq, PutRes](implicit
     } yield deferred.get.flatten
   }
 
+  /** Attempts to put a record into the producer's buffer, to be batched and
+    * produced at a defined interval.
+    *
+    * @param record
+    *   [[kinesis4cats.producer.Record Record]]
+    * @return
+    *   F of Option of F of Producer.Result. Inner F represents a `deferred.get`
+    *   call, which will complete when the record has been published. F[None]
+    *   means the producer queue is full or has been shut down.
+    */
+  def tryPut(record: Record): F[Option[F[Producer.Result[PutRes]]]] = {
+    val ctx = LogContext()
+
+    for {
+      _ <- logger.debug(ctx.context)("Received record to put")
+      deferred <- Deferred[F, F[Producer.Result[PutRes]]]
+      sendRes <- channel.trySend(record -> deferred)
+      res <- sendRes.fold(
+        _ =>
+          logger
+            .warn(ctx.context)(
+              "Producer has been shut down and will not accept further requests"
+            )
+            .as(none[F[Producer.Result[PutRes]]]),
```

Comment:
**thought:** tbh I would just raise in this case. The exception can still be transformed into a None on the caller side (if it's of a specific, public type)